### DISCUSSION TITLE
Get via CLI subapps openapi.json considering mount paths

### DISCUSSION THREAD

**Discussion by staticdev** [fastapi/fastapi#5589] [Questions] [answered]
When I have a simple app such as:

main.py

```python
import fastapi
app = fastapi.Fastapi()
```

I already get the whole openapi.json a app via CLI (CI/CD):

```sh
python -c 'from main import app; import json; print(json.dumps(app.openapi()))' > openapi.json
```

But when I use [sub-applications](https://fastapi.tiangolo.com/advanced/sub-applications/) and have let's say:

main.py

```python
import fastapi
import subapp

app = fastapi.Fastapi()
app.mount("/subapp", subapp.app)
```

I want to generate `openapi.json` just for this **subapp CONSIDERING the mount path**. I can download it on `/subapp/openapi.json`, but did not find a way to generate it via CLI. For instance,

```sh
python -c 'from subapp import app; import json; print(json.dumps(app.openapi()))' > openapi.json
```

This gives me an openapi.json that is different from the one from `/subapp/openapi.json` since it is not considering the mount path.

Any idea what would be the simplest way to do that and possibly not have to run the application @tiangolo ?

**Comment by tiangolo** [comment, marked as answer]
Not sure, maybe you can iterate over the mounts and extract the prefix, and then from that mount get the app and generate the OpenAPI and add the prefix to it.
